app/scripts/metamask-controller.js

```diff
@@ -2136,6 +2331,26 @@ export default class MetamaskController extends EventEmitter {
     return await promise;
   }
 
+  ///: BEGIN:ONLY_INCLUDE_IN(flask)
+  /**
+   * TODO:snaps verify safety of using keyringController.getAccounts
+   *
+   * @param domain
+   * @param requestedAccount
+   */
+  async getAppKeyForDomain(domain, requestedAccount) {
```

Comment:
Consider removing this and adding it later with a proper permission

Reply:
During the initial discussions about app keys ~2 years ago, we never intended for them to require a permission. See also: https://github.com/MetaMask/metamask-extension/pull/11837#discussion_r790304000